evalml/pipelines/components/component_base.py

```diff
@@ -103,13 +103,14 @@ def default_parameters(cls):
     def _supported_by_list_API(cls):
         return not cls.modifies_target
 
-    def _handle_partial_dependence_fast_mode(self, X, pipeline_parameters):
+    def _handle_partial_dependence_fast_mode(self, X, pipeline_parameters, target):
```

Comment:
Can we make `target` an optional parameter? I had assumed it was based on the usage in the fast mode utils, and I think it makes more sense since we don't need it in most cases.

Reply:
Yes, and I can do the same for `X` as well, since it's only needed for the dfs transformer as well,

Reply:
as more explanation: the reason why I hadn't originally made the extra args be optional is because we only call `_handle_partial_dependence_fast_mode` in one place where we have to pass in all the optional args. 

There's something that felt off about making the args optional when they will always be passed in and not be null. And because they're always passed in, we have to have them present in the component args even when they aren't used. I'd almost rather just use `_` in those components for the unused args, but I get that having them be optional makes them more flexible if we did want to use them in some other way in the future and is more technically correct, so I'm happy to make the change!